### PR TITLE
Added user model and settings page

### PR DESCRIPTION
**Pull request by andreaslordos**
# Diff Summary

## Context / Problems / Issues

Integrate CoreData into the application to persist user data securely. User model created during setup and maintained during application.

## This Diff / Solutions
This diff provides data model, and interactions with that model, needed to persist User data throughout the app (e.g. logged period events, etc.) as well as the UI layout for E2EE data migration for Hera

## Next Steps
Add backend logic for E2EE data migration

## Screenshots
**Data model**
Screen Shot 2022-07-28 at 4 01 44 PM

**Settings page**
Screen Shot 2022-07-28 at 4 03 07 PM

**QR transfer**
Screen Shot 2022-07-28 at 4 03 37 PM

### CODE REVIEW

File: Hera/Entities/Cycles/Cycle+CoreDataProperties.m [hunk 1]
```diff
@@ -0,0 +1,24 @@
+//
+//  Cycle+CoreDataProperties.m
```
Comment (zqiuuuu) [inline Hera/Entities/Cycles/Cycle+CoreDataProperties.m:2]: can we rename this file,  you can call it CycleAndCoreDataProperties.m instead

File: Hera/Entities/Events/Event+CoreDataClass.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//
```
Comment (zqiuuuu) [inline Hera/Entities/Events/Event+CoreDataClass.h:1]: ditto for the file name changes

File: Hera/Entities/Cycles/CycleCollectionPast+CoreDataClass.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
+//
```
Comment (zqiuuuu) [inline Hera/Entities/Cycles/CycleCollectionPast+CoreDataClass.h:1]: One idea to make those file name shorter for readability is to remove the prefix Cycle as they are already under the Cycle folder

File: Hera/ViewControllers/HeraTabBarController.m [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+    self.context = [Utilities getObjectContext]; // load object context only AFTER login / auth is completed.
+}
+
+/*
+#pragma mark - Navigation
+
+// In a storyboard-based application, you will often want to do a little preparation before navigation
```
Comment (zqiuuuu) [inline Hera/ViewControllers/HeraTabBarController.m:27]: remove if it's not be used?

**Review by zqiuuuu — APPROVED** [review]
let's have PR with less commits going forward

File: Hera/Entities/Cycles/CycleCollectionPast+CoreDataClass.h (hunk 3 above)
Comment (andreaslordos) [reply Hera/Entities/Cycles/CycleCollectionPast+CoreDataClass.h:1]: Anything ending in CoreDataClass and CoreDataProperties is auto-generated to conform to the naming convention of CoreData files

Comment (andreaslordos) [reply Hera/Entities/Cycles/CycleCollectionPast+CoreDataClass.h:1]: I could rename the class names themselves (e.g. CycleCollectionPast -> CollectionPast) but that would make the class name "CollectionPast" so nothing to indicate that it's a collection of cycles.. What do you think?

File: Hera/ViewControllers/HeraTabBarController.m (hunk 4 above)
Comment (andreaslordos) [reply Hera/ViewControllers/HeraTabBarController.m]: done

**Pull request merged**